src/newrelic/app_test.go

```diff
@@ -408,10 +414,9 @@ func TestConnectPayloadEncoded(t *testing.T) {
 		`"metadata":{"NEW_RELIC_METADATA_ONE":"one","NEW_RELIC_METADATA_TWO":"two"},` +
 		`"identifier":"one;two",` +
 		`"utilization":{"metadata_version":1,"logical_processors":22,"total_ram_mib":1000,"hostname":"some_host"},` +
-		`"event_harvest_config":{"report_period_ms":60000,` +
-		`"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,` +
-		`"span_event_data":` + strconv.Itoa(limits.MaxSpanMaxEvents) +
-		`,"log_event_data":` + strconv.Itoa(limits.MaxLogMaxEvents) + `}}` +
+		`"event_harvest_config":{"report_period_ms":60000,"harvest_limits":{"error_event_data":100,"analytic_event_data":10000,"custom_event_data":10000,` +
+		`"span_event_data":` + strconv.Itoa(limits.MaxSpanMaxEvents) + `,` +
+		`"log_event_data":` + strconv.Itoa(limits.MaxLogMaxEvents) + `}}` +
```

Comment:
There's some inconsistency how `expected` variable is constructed in this test. In some cases hardcoded values are used (`"span_event_data":2323`) and in some constants from `limits` package are used (`"span_event_data": + strconv.Itoa(limits.MaxSpanMaxEvents)`). Maybe it's not the place and time for this type of refactoring but wanted to point out that improvement is possible.

Reply:
Perhaps it would be good to have a constant for these cases where a valid limit is used which is different than an existing limit based constant.
I feel that is something we perhaps can do in a future general cleanup of these test cases.  There is a bunch of duplicated code where most of the expected JSON is the same and only a single field is changed, for example.  In processor_test.go I make a creator for a handling this which accepted a parameter for the changing element so there is less duplication and change of making a typo copying these long strings.